x-pack/solutions/observability/plugins/observability_logs_explorer/public/redirect_to_discover.tsx

```diff
@@ -27,7 +28,35 @@ export const renderDiscoverRedirect = (core: CoreStart, appParams: AppMountParam
 
 export const DiscoverRedirect = ({ core }: { core: CoreStart }) => {
   const location = useLocation();
-  const path = `${location.pathname}${location.search}`;
+  const searchParams = new URLSearchParams(location.search);
+  const pageStateEncoded = searchParams.get('pageState') || '';
+  const pageState = (safeDecode(pageStateEncoded) || {}) as {
+    time?: { from: string; to: string };
+    query?: { language: string; query: string };
+    filters?: any[];
+    columns?: any[];
+    refreshInterval?: { pause: boolean; value: number };
+  };
+
+  const gState = {
+    time: pageState.time || { from: 'now-15m', to: 'now' },
+    filters: [],
+    refreshInterval: pageState.refreshInterval || { pause: true, value: 60000 },
+  };
+
+  const aState = {
+    query: pageState.query,
+    filters: Array.isArray(pageState.filters) ? pageState.filters : [],
+    columns: Array.isArray(pageState.columns) ? pageState.columns : [],
+  };
+
+  const gEncoded = encode(gState);
+  const aEncoded = encode(aState);
+
+  const newSearch = `#/?_g=${gEncoded}&_a=${aEncoded}`;
+  const path = `${location.pathname}${newSearch}`;
```

Comment:
Nit: These query params are part of our public API so this is technically safe to do, although in general I'd recommend opting for the Discover locator whenever possible to avoid manually constructing them.

Reply:
Agreed - and although navigating like this directly in the component was already introduced in an earlier PR, I wonder if we shouldn't do this cleanly and use the proper locator in a proper effect. Does that make sense?

Reply:
It does on my end for sure!

Reply:
Only thing I was wondering if it would somehow impact the performance of a plugin. 
Correct me if I'm wrong but in order to use locator we would have to wrap the plugin with Kibana Context Provider. I just wasn't sure if it would affect the plugin anyhow

Reply:
valid point - but in order to access the locator you only need the `share` plugin, which you could pass to the `renderDiscoverRedirect` in the `plugin.ts` in addition to `coreStart`: https://github.com/elastic/kibana/blob/e0f106f1b7e34357e43a0c8819b52aa39134483c/x-pack/solutions/observability/plugins/observability_logs_explorer/public/plugin.ts#L38-L40

Reply:
oh that's the magic I didn't know 😄 
Will apply.

Reply:
I took the code from your comment,so it was definitely helpful 😃